caffe2/operators/elementwise_ops.h

```diff
@@ -176,17 +176,19 @@ class BinaryElementwiseWithArgsOp final : public Operator<Context> {
           A.sizes().cbegin(), A.sizes().cend(), std::back_inserter(A_dims));
       std::copy(
           B.sizes().cbegin(), B.sizes().cend(), std::back_inserter(B_dims));
-      const std::vector<int> C_dims =
+      auto C_dims_int =
           elementwise_ops_utils::ComputeBinaryBroadcastForwardDims(
               A_dims, B_dims);
-      if (C == &A) {
-        CAFFE_ENFORCE_EQ(C_dims, A_dims);
-      } else if (C == &B) {
-        CAFFE_ENFORCE_EQ(C_dims, B_dims);
-      } else {
-        C->Resize(C_dims);
+      std::copy(
```

Comment:
m we should probably change the return type of ComputeBinaryBroadcastForwardDims to vector<int64_t>, so this weird copy isn't necessary. Probably shouldn't be done in this diff though.

Reply:
yeah, there are some cases in "caffe2/utils/math.h" as well